File: recgame/src/main/java/com/br/recgame/model/Plataforma.java
package com.br.recgame.model;

import java.util.Date;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
public class Plataforma {
    @Id
    private Long idPlataforma;
    @Transient
    public final String SEQUENCIA = "plataforma_sequencia";
    private String descricao;
    private Date dataCadastro;

    public Plataforma() {}

    public Long getIdPlataforma() {
        return idPlataforma;
    }

    public void setIdPlataforma(Long idPlataforma) {
        this.idPlataforma = idPlataforma;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public Date getDataCadastro() {
        return dataCadastro;
    }

    public void setDataCadastro(Date dataCadastro) {
        this.dataCadastro = dataCadastro;
    }

    @Override
    public String toString() {
        return "Plataforma [dataCadastro=" + dataCadastro + ", descricao=" + descricao + ", idPlataforma="
                + idPlataforma + "]";
    }
}

File: flutter/rec_game_app/lib/models/plataforma.dart
class Plataforma {
  final int id;
  final String descricao;
  final String dataCadastro;
  final String avatar;

  const Plataforma({
    required this.id,
    required this.descricao,
    required this.dataCadastro,
    required this.avatar,
  });

  factory Plataforma.fromjson(Map<String, dynamic> json) {
    return Plataforma(
        id: json["idPlataforma"] as int,
        descricao: json["descricao"] as String,
        dataCadastro: json["dataCadastro"] as String,
        avatar: "");
  }
}
